plugin/gnome/NetworkManagerGnomeProxy.cpp

```diff
@@ -916,8 +916,17 @@ namespace WPEFramework
             }
             else if(ssid.ssid.size() <= 32)
             {
-                if(wifi->wifiConnect(ssid))
-                    rc = Core::ERROR_NONE;
+                if(ssid.passphrase.empty() && (ssid.security == Exchange::INetworkManager::WIFI_SECURITY_NONE)) 
```

Comment:
If `WIFIConnect()` is called for the first time but to connect to Open SSID (un-secure SSID) this API will fail as there is nothing to activate.

So, first we have to check whether the given SSID is in active connection list and then call for activate.